cEP-0005.md

```diff
@@ -0,0 +1,157 @@
+# coala Configuration
+
+| Metadata     |                                               |
+| ------------ |-----------------------------------------------|
+| cEP          | 5                                             |
+| Version      | 0.1                                           |
+| Title        | coala Configuration                           |
+| Authors      | Lasse Schuirmann <lasse.schuirmann@gmail.com> |
+| Status       | Proposed                                      |
+| Type         | Feature                                       |
+
+## Problem
+
+One main pain point for our users is that coala is hard to configure. It tries
+to provide full options to a full set of tools. At the same time we try to
+provide an abstraction over the tools by e.g. giving the same setting names to
+similar bears across languages so the user can reuse the same configuration for
+other languages too.
+
+By integrating lots of different existing tools with coala we have gained a
+huge set of possible analysis but it makes it very hard to understand and
+configure without knowledge about a particular tool. Also the conflict of bears
+versus linter bears that effectively offer multiple "checks" or sets of
+analyses rises: it is confusing to have to use multiple bears for one language
+for a set of analyses that is provided by only one bear for another language.
+
+Other problems of the configuration file are, that some duplication is
+involved, especially when dealing with settings that are used for file
+collection.
+
+## Proposal
+
+One important aspect of coala and it's usability is that the configuration of a
+new language is as easy as possible. There should be no learning involved.
+Naturally, one very good aspect of a good configuration would be that it can be
+used for a new language without changing anything. Also, namespaces are a
+honking great idea and explicit is better than implicit.
+
+The following changes are proposed and illustrated in the mockup following this
+paragraph:
+
+- Implicit `default` section inheritance will be removed. (It will be kept with
+  warning for some time to allow people to do the switch.)
+- Explicit inheritance is possible by giving sections namespaces (`all.python`
+  would inherit from the `all` section.)
+- Values can be added to inherited values. E.g. `ignore += ./vendor` would take
+  the inherited value and add `, ./vendor` to it. Note that additions within
+  a section will not be possible! The configuration should describe a state and
+  not a program and must remain flow insensitive.
+- To specify, what analysis should be run the user will not have to care about
+  bears. Instead, an `aspects` setting will be provided allowing the user to
```

Comment:
I think instead of detect and fix I'd rather improve the actions design because it's way more flexible, e.g. I can tell it to try fix it or to just show the patch and then ask me. @underyx has a WIP cEP for that.